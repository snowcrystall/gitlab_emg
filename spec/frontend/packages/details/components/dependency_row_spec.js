import { shallowMount } from '@vue/test-utils';
import DependencyRow from '~/packages/details/components/dependency_row.vue';
import { dependencyLinks } from '../../mock_data';

describe('DependencyRow', () => {
  let wrapper;

  const { withoutFramework, withoutVersion, fullLink } = dependencyLinks;

  function createComponent({ dependencyLink = fullLink } = {}) {
    wrapper = shallowMount(DependencyRow, {
      propsData: {
        dependency: dependencyLink,
      },
    });
  }

  const dependencyVersion = () => wrapper.find('[data-testid="version-pattern"]');
  const dependencyFramework = () => wrapper.find('[data-testid="target-framework"]');

  afterEach(() => {
    wrapper.destroy();
  });

  describe('renders', () => {
    it('full dependency', () => {
      createComponent();

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('version', () => {
    it('does not render any version information when not supplied', () => {
      createComponent({ dependencyLink: withoutVersion });

      expect(dependencyVersion().exists()).toBe(false);
    });

    it('does render version info when it exists', () => {
      createComponent();

      expect(dependencyVersion().exists()).toBe(true);
      expect(dependencyVersion().text()).toBe(fullLink.version_pattern);
    });
  });

  describe('target framework', () => {
    it('does not render any framework information when not supplied', () => {
      createComponent({ dependencyLink: withoutFramework });

      expect(dependencyFramework().exists()).toBe(false);
    });

    it('does render framework info when it exists', () => {
      createComponent();

      expect(dependencyFramework().exists()).toBe(true);
      expect(dependencyFramework().text()).toBe(`(${fullLink.target_framework})`);
    });
  });
});
