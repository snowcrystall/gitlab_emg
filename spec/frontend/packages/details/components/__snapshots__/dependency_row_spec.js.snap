// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyRow renders full dependency 1`] = `
<div
  class="gl-responsive-table-row"
>
  <div
    class="table-section section-50"
  >
    <strong
      class="gl-text-body"
    >
      Test.Dependency
    </strong>
     
    <span
      data-testid="target-framework"
    >
      (.NETStandard2.0)
    </span>
  </div>
   
  <div
    class="table-section section-50 gl-display-flex gl-md-justify-content-end"
    data-testid="version-pattern"
  >
    <span
      class="gl-text-body"
    >
      2.3.7
    </span>
  </div>
</div>
`;
